consumer.go

```diff
@@ -355,6 +361,24 @@ func (r *Consumer) ConnectToNSQLookupd(addr string) error {
 		}
 	}
 	r.lookupdHTTPAddrs = append(r.lookupdHTTPAddrs, parsedAddr)
+	if r.lookupdHttpClient == nil {
+		transport := &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout:   r.config.LookupdPollTimeout,
+				KeepAlive: r.config.LookupdPollAliveDuration,
```

Comment:
This is the TCP keepalive feature, not the http keepalive feature (which is enabled by default, just need to reuse the transport, instead of making a new one every time). This may not be needed:

https://pkg.go.dev/net#Dialer
```
	// KeepAlive specifies the interval between keep-alive
	// probes for an active network connection.
	// If zero, keep-alive probes are sent with a default value
	// (currently 15 seconds), if supported by the protocol and operating
	// system. Network protocols or operating systems that do
	// not support keep-alives ignore this field.
	// If negative, keep-alive probes are disabled.
	KeepAlive time.Duration
```

Reply:
hmm ... the `DefaultTransport` does still seem to explicitly configure this to 30 seconds ... we should probably similarly set it to 30 seconds, but not bother with making it configurable (it's the tcp layer thing, not http or nsq logic related, 30 seconds is probably fine everywhere, and the original problem was creating a new Transport for every request)